src/Columns/ColumnVector.cpp

```diff
@@ -135,6 +144,32 @@ void ColumnVector<T>::getPermutation(bool reverse, size_t limit, int nan_directi
     }
     else
     {
+
+        /// If bitonic sort if specified as preferred
+#if defined (BITONIC_SORT_PREFERRED) && defined (BITONIC_SORT_IMPL)
+        T* pairsData = new T[s];
```

Comment:
It's not OK to make new/delete ourself. If there would and exception inside we'll get memory leak here. Make an std::vector<T>, resize it by s, and use `data()` for row array.

Reply:
Resolved by using vectors and moving all this data manipulation to the method in BitonicSort class